client/app/bundles/course/assessment/submission/pages/SubmissionEditIndex/index.jsx

```diff
@@ -54,10 +64,42 @@ class VisibleSubmissionEditIndex extends Component {
     dispatch(autogradeSubmission(params.submissionId));
   }
 
+  /**
+   * Return true if it is ready to submit
+   */
+  validateSubmit = (showErrorNotification = true) => {
+    const { dispatch, form } = this.props;
+    const answers = Object.values(form.values);
+
+    // touch all the values
+    answers.forEach((answer = {}) => {
```

Comment:
I am not sure why you need to touch all answers here. I guess it's for validation? but should the answers be touched already when there's a change?  And if the answer is not touched, we should not validate it?

Reply:
And for validation, I think you should do it when the user trying to submit a single answer? otherwise, the user will still be able to submit empty answers in autograded assessment. And I am not sure if we should validate for manually graded assessment. 

I realised the issue #2507, so I'd suggest doing in this in a separate PR, and this might affect all question types too.

Reply:
> How about we name this validateFileType? I think that's less confusing and what this method does is actually validation.

Ok. Then I guess ```validateFile``` would be better. In the future if we can use it to validate file size etc.

> just a quick question, is there any case that when A is unmounted but B is not?

There is not any right now. But it does not mean this won't happen in the future. In the future we can use recording in any other components.

> I am not sure why you need to touch all answers here. I guess it's for validation?

Yes, it is for validation.

> but should the answers be touched already when there's a change? And if the answer is not touched, we should not validate it?

What if the user does not change the value, and press submit button directly? In this case we should still touch all the fields.

> And for validation, I think you should do it when the user trying to submit a single answer? otherwise, the user will still be able to submit empty answers in autograded assessment.

Yes, I have already added validation for submit a single answer. You may check ```handleSubmitAnswer``` 

> And I am not sure if we should validate for manually graded assessment.

In my opinion, we should treat manual grading and auto grading the same. Otherwise it will confuse the user. But you may discuss with the team.

Actually, touching all the values, should be done by redux-form automatically. We should pass ```onSubmit``` function to the form, and dispatch submit action. Adding click listener to the submit button is not the right way to use ```redux-form```. Otherwise I won't need to manually touch every fields. 

> I realised the issue #2507, so I'd suggest doing in this in a separate PR, and this might affect all question types too.

Ok, then I will remove the validation part from this PR.